addons/mail/static/src/core/common/thread_model.js

```diff
@@ -843,6 +843,11 @@ export class Thread extends Record {
             this.isLoadedDeferred.then(() => new Promise(setTimeout)).then(() => this.markAsRead());
         }
         if (this.selfMember) {
+            this.selfMember.localSeenMessage = newestPersistentMessage?.eq(
+                this.selfMember.seen_message_id
+            )
+                ? newestPersistentMessage
+                : this.selfMember.seen_message_id;
```

Comment:
As discussed, `localSeenMessage` shouldn't have its dynamic value being totally coupled with `newestPersistentMessage` and `selfMember.seen_message_id`.

Probably keep same `localSeenMessage` if it's already set. And have some code to clear it.